### PR TITLE
Add support for delete_factor and get_factor methods

### PR DESCRIPTION
**Pull request by awolfden**
- Add support for delete factor and get_factor methods
- Add tests for delete_factor and get_factor methods

### CODE REVIEW

File: tests/test_mfa.py [hunk 1]
```diff
@@ -169,34 +180,34 @@ def test_challenge_success(
         )
         assert challenge_factor == mock_challenge_factor_response
 
-    def test_verify_factor_no_id(self, mock_verify_challenge_payload):
+    def test_verify_challenge_no_id(self, mock_verify_challenge_payload):
         with pytest.raises(ValueError) as err:
-            self.mfa.verify_factor(
+            self.mfa.verify_challenge(
```
Comment (maxdeviant) [inline tests/test_mfa.py:184]: We want to keep the `verify_factor` tests in place to ensure the `verify_factor` method is still under test.

We already have test coverage for the `verify_challenge` method.

We can remove these tests when we remove the `verify_factor` method in the next major version.

File: workos/mfa.py [hunk 2]
```diff
@@ -88,6 +88,30 @@ def enroll_factor(
...
+
+        Returns: Does not provide a return.
+        """
+
+        if id is None:
+            raise ValueError("Incomplete arguments. Need to specify a factor id.")
+
+        response = self.request_helper.request(
+            "auth/factors/{id}",
+            method=REQUEST_METHOD_POST,
```
Comment (maxdeviant) [inline workos/mfa.py:109]: Should this be a DELETE?

```suggestion
            method=REQUEST_METHOD_DELETE,
```

File: tests/test_mfa.py (hunk 1 above)
Comment (awolfden) [reply tests/test_mfa.py]: That makes sense! I just removed those changes. ☑️

File: workos/mfa.py [hunk 3]
```diff
@@ -88,6 +89,59 @@ def enroll_factor(
 
         return WorkOSAuthenticationFactorSms.construct_from_response(response).to_dict()
 
+    def get_factor(
+        self,
+        id=None,
+    ):
+        """
+        Returns an authorization factor from it's ID.
```
Comment (maxdeviant) [inline workos/mfa.py:97]: Possessive "its" does not use an apostrophe:

```suggestion
        Returns an authorization factor from its ID.
```

File: workos/mfa.py [hunk 4]
```diff
@@ -88,6 +89,59 @@ def enroll_factor(
...
+
+        Kwargs:
+            id (str) - The id of the factor to be obtained.
```
Comment (maxdeviant) [inline workos/mfa.py:100]: nit: casing:

```suggestion
            id (str) - The ID of the factor to be obtained.
```

File: workos/mfa.py [hunk 5]
```diff
@@ -88,6 +89,59 @@ def enroll_factor(
...
+        Deletes an MFA authorization factor.
+
+        Kwargs:
+            id (str) - The id of the authorization factor to be deleted.
+
+        Returns: Does not provide a return.
+        """
+
+        if id is None:
+            raise ValueError("Incomplete arguments. Need to specify a factor id.")
```
Comment (maxdeviant) [inline workos/mfa.py:135]:
```suggestion
            raise ValueError("Incomplete arguments. Need to specify a factor ID.")
```

File: workos/mfa.py [hunk 6]
```diff
@@ -88,6 +89,59 @@ def enroll_factor(
...
+        if id is None:
+            raise ValueError("Incomplete arguments. Need to specify a factor ID.")
+
+        response = self.request_helper.request(
+            "auth/factors/{id}",
+            method=REQUEST_METHOD_DELETE,
+            token=workos.api_key,
+        )
+
+        return response
```
[the remark this answers is not in the document]
Comment (awolfden) [reply workos/mfa.py:143]: Removed the response ✅

File: workos/mfa.py [hunk 7]
```diff
@@ -88,6 +89,59 @@ def enroll_factor(
 
         return WorkOSAuthenticationFactorSms.construct_from_response(response).to_dict()
 
+    def get_factor(
+        self,
+        id=None,
```
Comment (blairlunceford) [inline workos/mfa.py:94]: Can we rename this `authentication_factor_id` to match the other methods? (also applies to the `delete_factor` method)

File: workos/mfa.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (blairlunceford) [reply workos/mfa.py:143]: It looks like the response is still there.

You can use this method as a guid for what to return here: https://github.com/workos/workos-python/blob/main/workos/sso.py#L252

[the remark this answers is not in the document]
Comment (awolfden) [reply workos/mfa.py]: Updated

File: workos/mfa.py (hunk 7 above)
Comment (awolfden) [reply workos/mfa.py]: Updated

**Review by blairlunceford — APPROVED** [review]
🎉

**Pull request merged**